cadl-tests/src/main/java/com/cadl/paged/PagedClient.java

```diff
@@ -61,4 +62,21 @@ public final class PagedClient {
     public PagedIterable<BinaryData> list(RequestOptions requestOptions) {
         return new PagedIterable<>(this.client.list(requestOptions));
     }
+
+    /**
+     * List Resource resources.
+     *
+     * @throws com.azure.core.exception.HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return paged collection of Resource items as paginated response with {@link PagedIterable}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedIterable<Resource> list() {
+        // Generated convenience method for list
+        return new PagedIterable<>(client.list());
```

Comment:
At present, it is fine to just call async method and convert to sync (if we do not need `Context`).

However, it could change when we had dedicated call stack for sync method.

Reply:
Similar to current convenience method on simple method, `Context` overload is not generated.

Reply:
Because it is pretty hard to convert `PagedIterable<BinaryData>` to `PagedIterable<Foo>`. The `mapPage` in core is buggy (actually `PagedFlux.mapPage` is also buggy, so we had another Utils class for these in mgmt https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/PagedConverter.java).